src/mapping/RadialBasisFctSolver.h

```diff
@@ -0,0 +1,181 @@
+#include "impl/BasisFunctions.hpp"
+#include "precice/types.hpp"
+#include "utils/EigenHelperFunctions.hpp"
+#include "utils/Event.hpp"
+
+namespace precice {
+extern bool syncMode;
+
+struct RadialBasisFctSolver {
+public:
+  RadialBasisFctSolver() = default;
+
+  /// Assembles the system matrices and computes the decomposition of the interpolation matrix
+  template <typename RADIAL_BASIS_FUNCTION_T>
+  void computeDecomposition(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, const mesh::Mesh &outputMesh, std::vector<bool> deadAxis);
```

Comment:
`_matrixA` and `_qr` are created here and without them, this class cannot do much.

So, isn't this function really the constructor of this class?

If this is the case, doesn't this type really represent the `Decomposition` itself?

Reply:
> Yes, you can see it as a constructor.

If the type provdes default construction and copy assignment, then you could write

```cpp
_rbfSolver = RadialBasicFctSolver{_basisFunction, _inputMesh, _outputMesh, _deadAxis};
```

( Yes, template deduction in constructors is another good reason to move to C++17 )

> Which type are you referring to?

The `RadialBasisFctSolver` really is a `RBFDecomposition`, right?

Reply:
You could also create an instance of `RadialBasicFctSolver` in `computeDecomposition`, essentially turning it into a factory method.

```cpp
_rbfSolver = computeDecomposition(_basisFunction, _inputMesh, _outputMesh, _deadAxis);
```

The useless default state is kind of covered by the assertions of Eigen, right?

Reply:
> The useless default state is kind of covered by the assertions of Eigen, right?

A bit cryptic to track down, but yes. I will add another assertion in the solve subroutine.

Reply:
I went now for the two constructor solution you proposed (although we still end up having a useless default state, which is asserted, however).